src/test/java/springboot/controller/CustomerControllerTest.java

```diff
@@ -0,0 +1,110 @@
+import org.springframework.jdbc.datasource.init.DatabasePopulator;
+import org.springframework.jdbc.datasource.init.DatabasePopulatorUtils;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import springboot.entity.Customer;
+
+import javax.sql.DataSource;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+/**
+ * Created by Danylo on 02.04.2019
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@TestPropertySource("classpath:test.properties")
+public class CustomerControllerTest {
+    private static final int STATUS_OK = 200;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    private DataSource dataSource;
+
+    private MockMvc mockMvc;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Before
+    public void setup() {
+        Resource initSchema = new ClassPathResource("scripts\\schema.sql");
+        Resource data = new ClassPathResource("scripts\\data.sql");
+        DatabasePopulator databasePopulator = new ResourceDatabasePopulator(initSchema, data);
+        DatabasePopulatorUtils.execute(databasePopulator, dataSource);
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+    }
+
+    @Test
+    public void testGetAllCustomers() throws Exception {
+        MvcResult mvcResult = mockMvc.perform(get("/customers")).andDo(print()).andReturn();
+        assertEquals(STATUS_OK, mvcResult.getResponse().getStatus());
+        assertEquals("application/json;charset=UTF-8", mvcResult.getResponse().getContentType());
+        List<Customer> orders = getListCustomerResult(mvcResult);
+        assertEquals(2, orders.size());
+    }
+
+    @Test
+    public void testGetCustomerByIdExist() throws Exception {
+        MvcResult mvcResult = mockMvc.perform(get("/customers/{id}", "1111")).andDo(print()).andReturn();
+        Customer customer = mapper.readValue(mvcResult.getResponse().getContentAsString(), Customer.class);
+        assertEquals(STATUS_OK , mvcResult.getResponse().getStatus());
+        assertNotNull(customer);
+    }
+
+    @Test
+    public void testGetCustomerByIdNotExist() throws Exception {
+        MvcResult mvcResult = mockMvc.perform(get("/customers/{id}", "-1")).andReturn();
+        assertEquals(STATUS_OK, mvcResult.getResponse().getStatus());
```

Comment:
Is it a correct test? How you can get status 200 OK, if you are checking NOT EXIST case?